server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/AbstractRangeAggregatorFactory.java

```diff
@@ -42,20 +46,44 @@
     private final InternalRange.Factory<?, ?> rangeFactory;
     private final R[] ranges;
     private final boolean keyed;
+    private final String aggregationTypeName;
 
+    public static void registerAggregators(ValuesSourceRegistry valuesSourceRegistry, String aggregationName) {
+        valuesSourceRegistry.register(aggregationName,
+            List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.DATE, CoreValuesSourceType.BOOLEAN),
```

Comment:
Is Boolean range a valid concept?

Reply:
That's a really good question.  I think master will let you do this, which is why I included it, but I agree it's kind of a nonsense operation. My strategy so far has been to keep these things supported for now, and after we merge have a conversation about what we want to do long term with these cases.